ultra/ultra/baselines/adapter.py

```diff
@@ -19,79 +19,144 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
-import math
-import random
-from sys import path
-
 import numpy as np
+import torch
 from scipy.spatial import distance
+import random, math, gym
+from sys import path
+from collections import OrderedDict
+from ultra.baselines.common.baseline_state_preprocessor import BaselineStatePreprocessor
+from ultra.baselines.common.social_vehicle_config import get_social_vehicle_configs
+from ultra.baselines.common.yaml_loader import load_yaml
 
 path.append("./ultra")
 from ultra.utils.common import (
-    ego_social_safety,
     get_closest_waypoint,
     get_path_to_goal,
+    ego_social_safety,
     rotate2d_vector,
 )
 
 seed = 0
 random.seed(seed)
 num_lookahead = 100
 
-# def action_adapter(agent.id):
-#     agent.action_space_type
-#     pass
-
 
 class BaselineAdapter:
-    def __init__(self):
-        pass
-
-    def observation_adapter(self, env_observation):
-        ego_state = env_observation.ego_vehicle_state
-        start = env_observation.ego_vehicle_state.mission.start
-        goal = env_observation.ego_vehicle_state.mission.goal
-        path = get_path_to_goal(
-            goal=goal, paths=env_observation.waypoint_paths, start=start
+    def __init__(self, agent_type):
```

Comment:
The agent type is there to get the social vehicle information from respective policy params.yaml files, which is then passed on to the state preprocessor. The reason we have it there as it for now, because there was other tasks depending on it. In the future, we can somehow refactor it so that we only need to supply the policy_class instead of the agent_type from the agent_spec